test/blackbox-tests/test-cases/pkg/convert-opam-commands.t

```diff
@@ -214,7 +214,7 @@ Package which has boolean where string was expected. This should be caught while
      (run echo g))
     (when
      (and
-      %{pkg-self:with-test}
+      %{with-test}
```

Comment:
Is there no analogous test for `with-doc`?

Reply:
I could add one, I wasn't sure if we wanted to go ahead with this.

Reply:
It's worth improving the test to include this.

Reply:
I won't improve this test, but the other opam var tests I added do test for that. I've rebased this PR now so you should see we are tested both with-doc and with-test properly.